sdk/storage/Azure.Storage.Blobs/src/LeaseClient.cs

```diff
@@ -245,12 +250,13 @@ private void EnsureClient()
         /// a failure occurs.
         /// </remarks>
         private async Task<Response<Lease>> AcquireInternal(
-            int duration,
+            TimeSpan duration,
             HttpAccessConditions? httpAccessConditions,
             bool async,
             CancellationToken cancellationToken)
         {
             this.EnsureClient();
+            var serviceDuration = duration < TimeSpan.Zero ? Constants.Blob.Lease.InfiniteLeaseDuration : Convert.ToInt64(duration.TotalSeconds);
```

Comment:
Add a comment about Int64 being an overflow safe cast relative to TimeSpan.MaxValue